sdk/src/lib.rs

```diff
@@ -488,6 +488,9 @@ impl ActivityHalf {
                         Ok(res) => ActivityExecutionResult::ok(res),
                         Err(err) => match err.downcast::<ActivityCancelledError>() {
                             Ok(ce) => ActivityExecutionResult::cancel_from_details(ce.details),
+                            Err(other_err) if other_err.is::<ActivityResultPendingError>() => {
```

Comment:
btw, I also snuck this in for activities to begin async completion.

Reply:
I saw :) -- I'm thinking I might actually want this to be a success value, like continue-as-new is for workflows. Since this really isn't an "error" state, I think it's semantically better to have it be a non-error value as long as the ergonomics remain good.

WDYT? The workflow result seems pretty reasonable right now, you just call `into` before returning in the normal path. I think that can work here too.

Reply:
I'm assuming you're thinking of making an explicit enum struct like ```WfExitValue<T>``` for activities right?  I know it uses derive_more for the happy path convenient ```into()```.  Nice trick ignoring all the other enum variants. 

Just so we're clear, you are suggesting we introduce the ```anyhow::Result<ActExitValue<T>>``` as the ActivityFunc return type, allowing the SDK to handle this better and also make the implementer of the ActivityFunc return this new enum explicitly instead of using a magical ```Err(ActivityCancelledError)``` ?  If so, this is the right way to go, but also causes a big change.  I implemented it this way because the go sdk does this (which is also odd), it's quick and dirty, and also doesn't affect/break the API.

I'm for it though.  This should suffice: 
```
#[derive(Debug, derive_more::From)]
pub enum ActExitValue<T: Debug> {
    /// Completion requires an asynchronous callback
    WillCompleteAsync,
    /// Finish with a result
    Normal(T),
}
```
and the rest is just cleanup in the tests.  Give me the green light and I can take a crack at it.

Reply:
Yep, that's exactly what I mean. I would love it if you made that change. I think it's potentially quite confusing to new users to see this as an "error" value.

The blast radius is nonzero though indeed.

One thing that would be worth playing with, which I haven't been able to try yet, is if we could possibly avoid needing the explicit `into` call as well, by doing something like this (also reduces the blast radius, which is great):

Change:
```rust
impl<A, Rf, R, F> IntoActivityFunc<A, Rf> for F
where
    F: (Fn(ActContext, A) -> Rf) + Sync + Send + 'static,
    A: FromJsonPayloadExt + Send,
    Rf: Future<Output = Result<R, anyhow::Error>> + Send + 'static,
    R: AsJsonPayloadExt,
```

to

```rust
impl<A, Rf, R, F> IntoActivityFunc<A, Rf> for F
where
    F: (Fn(ActContext, A) -> Rf) + Sync + Send + 'static,
    A: FromJsonPayloadExt + Send,
    Rf: Future<Output = Result<RR, anyhow::Error>> + Send + 'static,
    R: AsJsonPayloadExt,
    RR: Into<ActExitValue<R>>
```

If that works, that'd be real nice

Reply:
Implemented this in the latest commit and removed the pending error.  I had to introduce the type parameter into ```IntoActivityFunc``` trait though, but otherwise your new trait gymnastics works well.  Take a look and let me know what you think.  Everything else remains unchanged because of this impl.